framework/src/actions/CreateProblemAction.C

```diff
@@ -68,6 +69,28 @@ CreateProblemAction::act()
   {
     // build the problem only if we have mesh
     {
+      // if users try to use Eigen Executioner and do not specify a problem type,
+      // we should create EigenProblem
+      if (_app.useEigenExecutioner() && _type == "FEProblem")
+      {
+        _type = "EigenProblem";
+        parameters().set<std::string>("type") = "EigenProblem";
+        _moose_object_pars =
+            (!parameters().have_parameter<bool>("skip_param_construction") ||
+                     (parameters().have_parameter<bool>("skip_param_construction") &&
+                      !parameters().get<bool>("skip_param_construction"))
+                 ? _factory.getValidParams(_type)
+                 : validParams<MooseObject>());
+        // if there is a [Problem] block, the parameters will be retrieved from parser
+        if (parameters().isParamValid("parser_syntax"))
+        {
+          _app.parser().extractParams(parameters().get<std::string>("parser_syntax"),
```

Comment:
Note that I dislike invoking the parser directly (as we've already discussed, I'll take a look at this and see if I can come up with a better way).